pymeasure/instruments/ni/virtualbench.py

```diff
@@ -0,0 +1,1713 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2019 PyMeasure Developers
+# pyvirtualbench library: Copyright (c) 2015 Charles Armstrap <charles@armstrap.org>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import re
+# ctypes only required for VirtualBench_Direct class
+from ctypes import (c_bool, c_size_t, c_double, c_uint8, c_int32, c_uint32,
+                    c_int64, c_uint64, c_wchar, c_wchar_p, Structure, c_int,
+                    cdll, byref)
+from datetime import datetime, timezone, timedelta
+import numpy as np
+import pandas as pd
+import pyvirtualbench as pyvb
```

Comment:
This will need to be included as an optional dependency. It's worth doing a try-except around this import and giving a sensible warning using the log.

Reply:
I implemented a try-except statement which emits a warning to the logger. So far I only catched the ModuleNotFoundError. I'm unsure whether it makes sense to catch more generously...